app/src/main/java/com/xa/xpensauditor/Day_wise_visualization.java

```diff
@@ -0,0 +1,204 @@
+package com.xa.xpensauditor;
+
+import static java.lang.System.currentTimeMillis;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.recyclerview.widget.RecyclerView;
+
+import android.os.Bundle;
+import android.util.Log;
+import android.widget.Button;
+import android.widget.DatePicker;
+import android.widget.TextView;
+
+import com.anychart.AnyChart;
+import com.anychart.AnyChartView;
+import com.anychart.chart.common.dataentry.DataEntry;
+import com.anychart.chart.common.dataentry.ValueDataEntry;
+import com.anychart.charts.Cartesian;
+import com.anychart.core.cartesian.series.Column;
+import com.anychart.enums.Anchor;
+import com.anychart.enums.HoverMode;
+import com.anychart.enums.Position;
+import com.anychart.enums.TooltipPositionMode;
+import com.firebase.client.Firebase;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.ValueEventListener;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Day_wise_visualization extends AppCompatActivity {
+    private Firebase mRootRef;
+    private Firebase RefUid,RefTran,RefUid1;
+    private Firebase RefName,RefEmail,RefPhnnum;
+    private Firebase RefCat,RefFood,RefHealth,RefTravel,RefEdu,RefBills,RefHomeNeeds,RefOthers,RefUncat;
+    private List<Transaction> transList = new ArrayList<>();
+    private RecyclerView recyclerView;
+    private TransactionAdapter mAdapter;
+    private DatePicker dateTransac1;
+    private DatePicker dateTransac2;
+    private Button Vis;
+    private String Tid;
+    int tot=0;
+
+    static Map<Integer, Integer> hm = new HashMap<>();
+
+    String tot1="Yes";
+    FirebaseAuth auth;
+    String Uid;
+    String day1, month1, year1;
+    String day2, month2, year2;
+    int d1, m1, y1;
+    int d2, m2, y2;
+    private Firebase RefTran3;
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_day_wise_visualization);
+        auth = FirebaseAuth.getInstance();
+        mRootRef=new Firebase("https://xpensauditor-g11-default-rtdb.firebaseio.com/");
+        mRootRef.keepSynced(true);
+        Uid=auth.getUid();
+        RefUid= mRootRef.child(Uid);
+        RefName = RefUid.child("Name");
+        RefEmail=RefUid.child("Email");
+
+
+        // RefUid.child("DateRange").child(String.valueOf(month + "-" + year)).child("Transactions").child(Tid).child("Amount").setValue(Amount);
+        Bundle bundle = getIntent().getExtras();
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def' child has incorrect indentation level 8, expected level should be 4.